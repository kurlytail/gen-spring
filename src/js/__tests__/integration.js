import { execSync } from 'child_process';

describe('# integration test', () => {
    beforeEach(() => {
        execSync('rm -rf testoutput');
        execSync('mkdir testoutput');
        execSync('git init', { cwd: 'testoutput' });
        execSync('git config user.email "you@example.com"', { cwd: 'testoutput' });
        execSync('git config user.name "Your Namer"', { cwd: 'testoutput' });
        execSync('git commit --allow-empty -m "Empty commit."', { cwd: 'testoutput' });
    });

    it('## should generate design and run spring commands', () => {
        let output = execSync('npm run build').toString();
        output = execSync('sgen -g `pwd`/dist/spring.min.js -d src/test/fixture/design.json -o testoutput').toString();
        output = output.replace(/info: Loaded generator .*spring.min.js.*/, '');
        output = output
            .replace(/warn: Please cherrypick changes from master-sgen-generated from .*/, '')
            .replace(/info: git cherry-pick .*/, '');
        expect(output).toMatchSnapshot();
        execSync('npm install', { cwd: 'testoutput', stdio: 'inherit' });
        execSync('npm run lint', { cwd: 'testoutput', stdio: 'inherit' });
        execSync('npm run build', { cwd: 'testoutput', stdio: 'inherit' });
    });
});
