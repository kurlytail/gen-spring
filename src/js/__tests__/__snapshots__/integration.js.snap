// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# integration test ## should generate design 1`] = `
"info: loading map node_modules/@kurlytail/gen-pom/templates/map.json
info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.settings/org.eclipse.jdt.core.prefs from template node_modules/@kurlytail/gen-pom/templates/.settings/org.eclipse.jdt.core.prefs._
info: Generated testoutput/.settings/org.eclipse.m2e.core.prefs from template node_modules/@kurlytail/gen-pom/templates/.settings/org.eclipse.m2e.core.prefs._
info: Generated testoutput/src/main/java/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/main/java/.gitkeep._
info: Generated testoutput/src/main/resources/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/main/resources/.gitkeep._
info: Generated testoutput/src/test/java/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/test/java/.gitkeep._
info: Generated testoutput/src/test/resources/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/test/resources/.gitkeep._
info: Generated testoutput/.classpath from template node_modules/@kurlytail/gen-pom/templates/.classpath._
info: Generated testoutput/.project from template node_modules/@kurlytail/gen-pom/templates/.project._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-model-version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-group-id._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-artifact-id._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-name._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-description._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-maven-surefire-plugin.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-junit-platform.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-junit-jupiter.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-java.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter-engine.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter-api.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-plugins-maven-surefire-plugin.plugin._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-plugins-maven-compiler-plugin.plugin._
info: Generated testoutput/pom.xml from template node_modules/@kurlytail/gen-pom/templates/pom.xml._
info: Generated testoutput/.gitignore from template node_modules/@kurlytail/gen-pom/templates/.gitignore._
info: Generated testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-pom/templates/Jenkinsfile._
info: Generated testoutput/settings.xml from template node_modules/@kurlytail/gen-pom/templates/settings.xml._
info: Generated testoutput/maven.pem from template node_modules/@kurlytail/gen-pom/templates/maven.pem._
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design and run spring commands 1`] = `
"info: loading map node_modules/@kurlytail/gen-pom/templates/map.json
info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.settings/org.eclipse.jdt.core.prefs from template node_modules/@kurlytail/gen-pom/templates/.settings/org.eclipse.jdt.core.prefs._
info: Generated testoutput/.settings/org.eclipse.m2e.core.prefs from template node_modules/@kurlytail/gen-pom/templates/.settings/org.eclipse.m2e.core.prefs._
info: Generated testoutput/src/main/java/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/main/java/.gitkeep._
info: Generated testoutput/src/main/resources/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/main/resources/.gitkeep._
info: Generated testoutput/src/test/java/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/test/java/.gitkeep._
info: Generated testoutput/src/test/resources/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/test/resources/.gitkeep._
info: Generated testoutput/.classpath from template node_modules/@kurlytail/gen-pom/templates/.classpath._
info: Generated testoutput/.project from template node_modules/@kurlytail/gen-pom/templates/.project._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-model-version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-group-id._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-artifact-id._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-name._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-description._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-maven-surefire-plugin.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-junit-platform.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-junit-jupiter.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-java.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter-engine.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter-api.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-plugins-maven-surefire-plugin.plugin._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-plugins-maven-compiler-plugin.plugin._
info: Generated testoutput/pom.xml from template node_modules/@kurlytail/gen-pom/templates/pom.xml._
info: Generated testoutput/.gitignore from template node_modules/@kurlytail/gen-pom/templates/.gitignore._
info: Generated testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-pom/templates/Jenkinsfile._
info: Generated testoutput/settings.xml from template node_modules/@kurlytail/gen-pom/templates/settings.xml._
info: Generated testoutput/maven.pem from template node_modules/@kurlytail/gen-pom/templates/maven.pem._
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design with merge 1`] = `
"info: loading map node_modules/@kurlytail/gen-pom/templates/map.json
info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.settings/org.eclipse.jdt.core.prefs from template node_modules/@kurlytail/gen-pom/templates/.settings/org.eclipse.jdt.core.prefs._
info: Generated testoutput/.settings/org.eclipse.m2e.core.prefs from template node_modules/@kurlytail/gen-pom/templates/.settings/org.eclipse.m2e.core.prefs._
info: Generated testoutput/src/main/java/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/main/java/.gitkeep._
info: Generated testoutput/src/main/resources/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/main/resources/.gitkeep._
info: Generated testoutput/src/test/java/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/test/java/.gitkeep._
info: Generated testoutput/src/test/resources/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/test/resources/.gitkeep._
info: Generated testoutput/.classpath from template node_modules/@kurlytail/gen-pom/templates/.classpath._
info: Generated testoutput/.project from template node_modules/@kurlytail/gen-pom/templates/.project._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-model-version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-group-id._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-artifact-id._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-name._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-description._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-maven-surefire-plugin.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-junit-platform.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-junit-jupiter.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-java.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter-engine.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter-api.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-plugins-maven-surefire-plugin.plugin._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-plugins-maven-compiler-plugin.plugin._
info: Generated testoutput/pom.xml from template node_modules/@kurlytail/gen-pom/templates/pom.xml._
info: Generated testoutput/.gitignore from template node_modules/@kurlytail/gen-pom/templates/.gitignore._
info: Generated testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-pom/templates/Jenkinsfile._
info: Generated testoutput/settings.xml from template node_modules/@kurlytail/gen-pom/templates/settings.xml._
info: Generated testoutput/maven.pem from template node_modules/@kurlytail/gen-pom/templates/maven.pem._
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design with merge 2`] = `
"info: loading map node_modules/@kurlytail/gen-pom/templates/map.json
info: loading map templates/map.json
info: Opening git repository testoutput/.git
warn: Ignoring error Error: cannot stash changes - there is nothing to stash.
info: Attempting to use branch master-sgen-generated
warn: Checkout branch master-sgen-generated
info: Generated testoutput/.settings/org.eclipse.jdt.core.prefs from template node_modules/@kurlytail/gen-pom/templates/.settings/org.eclipse.jdt.core.prefs._
info: Generated testoutput/.settings/org.eclipse.m2e.core.prefs from template node_modules/@kurlytail/gen-pom/templates/.settings/org.eclipse.m2e.core.prefs._
info: Generated testoutput/src/main/java/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/main/java/.gitkeep._
info: Generated testoutput/src/main/resources/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/main/resources/.gitkeep._
info: Generated testoutput/src/test/java/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/test/java/.gitkeep._
info: Generated testoutput/src/test/resources/.gitkeep from template node_modules/@kurlytail/gen-pom/templates/src/test/resources/.gitkeep._
info: Generated testoutput/.classpath from template node_modules/@kurlytail/gen-pom/templates/.classpath._
info: Generated testoutput/.project from template node_modules/@kurlytail/gen-pom/templates/.project._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-model-version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-group-id._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-artifact-id._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-name._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-description._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-maven-surefire-plugin.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-junit-platform.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-junit-jupiter.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-property-java.version._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter-engine.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-dependencies-junit-jupiter-api.dependency._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-plugins-maven-surefire-plugin.plugin._
info: loading extension node_modules/@kurlytail/gen-pom/templates/extensions/pom-plugins-maven-compiler-plugin.plugin._
info: Generated testoutput/pom.xml from template node_modules/@kurlytail/gen-pom/templates/pom.xml._
info: Generated testoutput/.gitignore from template node_modules/@kurlytail/gen-pom/templates/.gitignore._
info: Generated testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-pom/templates/Jenkinsfile._
info: Generated testoutput/settings.xml from template node_modules/@kurlytail/gen-pom/templates/settings.xml._
info: Generated testoutput/maven.pem from template node_modules/@kurlytail/gen-pom/templates/maven.pem._
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
warn: Checkout branch master-sgen-generated
warn: Checkout branch master-sgen-generated
warn: Nothing changed in generated files
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should print help options 1`] = `
"Usage:
  node sgen [OPTION]

Options:
  -m, --map=ARG+       mapping files
  -d, --design=ARG+    design files
  -e, --extension=ARG+ extension directories or files
  -o, --output=ARG     output directory for generated files
      --overwrite=ARG  Overwrite modes, can be one of skip, regen, error                                  (default: error)
      --forceOverwrite Override overwrite flags in the generator map with the one provided on --overwrite
  -h, --help           display this help
"
`;
